### DISCUSSION TITLE
startup code

### DISCUSSION THREAD

**Discussion by schoeberl** [lf-lang/lingua-franca#2206] [Ideas]
Startup code should not be part of the mission phase. Startup code might be long-running code and can delay periodic timers too much that have an offset of 0. Therefore, logic time shall be synchronized with physical time after all startup reactions have been executed.

**Comment by edwardalee** [comment]
I think this is application specific.  Timers can be given an offset greater than 0, so is this really an issue?

**Comment by schoeberl** [comment]
I see two issues: (1) we need to know the maximum execution time of the setup code and (2) every periodic timer in the application (can be a deeply embedded sub component) needs to encode this maximum execution time.I think real-time systems have a notion of setup time before mission start.

**Comment by lhstrh** [comment]
Just to clarify, what do you mean by "startup code?" Is it reactions that have a `startup` trigger?

**Comment by schoeberl** [comment]
Yes, exactly.
